From 7fe2a9bb56e1d8a0f9e7db9f8c46fad6c59ce860 Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Thu, 5 Mar 2020 09:34:43 +0800
Subject: [PATCH] dt-bindings: sound: rockchip: Add binding for rockchip codec
 digital

It's mainly composed of digital ADC and digital DAC. The digital ADC
converts PDM data into PCM data and then sends them out through I2S.
The digital DAC receives PCM data from I2S and then converts them into
PDM data.

Change-Id: I46533228b67c127e6fa70d45d4152763fa7290e0
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 .../bindings/sound/rockchip,codec-digital.txt | 31 +++++++++++++++++++
 1 file changed, 31 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/rockchip,codec-digital.txt

diff --git a/Documentation/devicetree/bindings/sound/rockchip,codec-digital.txt b/Documentation/devicetree/bindings/sound/rockchip,codec-digital.txt
new file mode 100644
index 000000000000..e66aa3760538
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/rockchip,codec-digital.txt
@@ -0,0 +1,31 @@
+* Rockchip Codec Digital Interface
+
+Required properties:
+
+- compatible: should be one of the following
+   - "rockchip,codec-digital-v1"
+   - "rockchip,rv1126-codec-digital"
+- reg: physical base address of the controller and length of memory mapped
+  region.
+- clocks: a list of phandle + clock-specifer pairs, one for each entry in clock-names.
+- clock-names: clock names.
+- rockchip,bclk-fs: configure the bclk fs.
+- resets: a list of phandle + reset-specifer paris, one for each entry in reset-names.
+- reset-names: reset names, should include "reset".
+- rockchip,grf: the phandle of the syscon node for GRF register.
+
+Example:
+
+rkacdc_dig: codec-digital@ff850000 {
+	compatible = "rockchip,codec-digital-v1";
+	reg = <0xff850000 0x1000>;
+	clocks = <&cru CLK_ACDCDIG_ADC>, <&cru CLK_ACDCDIG_DAC>,
+		 <&cru PCLK_ACDCDIG>;
+	clock-names = "adc", "dac", "pclk";
+	pinctrl-names = "default";
+	pinctrl-0 = <&acodec_pins>;
+	resets = <&cru SRST_ACDCDIG>;
+	reset-names = "reset" ;
+	rockchip,grf = <&grf>;
+	status = "disabled";
+};
-- 
2.35.3

